app/src/main/java/com/kickstarter/viewmodels/projectpage/LatePledgeCheckoutViewModel.kt

```diff
@@ -203,34 +211,56 @@ class LatePledgeCheckoutViewModel(val environment: Environment) : ViewModel() {
     }
 
     fun onPledgeButtonClicked(selectedCard: StoredCard?) {
+        // - Avoid launching any other coroutine call while previous coroutine active
+        if (pledgeButtonClickedJob?.isActive.isTrue()) return
+
         this.pledgeData?.let {
-            val project = it.projectData().project()
-            createPaymentIntentForCheckout(selectedCard, project, it.checkoutTotalAmount())
+            pledgeButtonClickedJob = viewModelScope.launch {
+                val project = it.projectData().project()
+
+                buttonEnabled = false
+                emitCurrentState(isLoading = true)
+
+                if (paymentIntent == null)
+                    createPaymentIntentForCheckout(selectedCard, project, it.checkoutTotalAmount())
+
+                else {
+                    paymentIntent?.let { pi ->
+                        selectedCard?.let { card ->
+                            validateCheckout(
```

Comment:
After failed by user 3DS challenge, call `validateCheckout` with stored in memory paymentIntent